### PR TITLE
Add Epistemic strategy to DeepEdit App

### PR DESCRIPTION
**Pull request by diazandr3s**
Signed-off-by: Andres <diazandr3s@gmail.com>

- Added dropout argument in init method in DynUNet (https://github.com/Project-MONAI/MONAI/pull/2947)
- Epistemic strategy woking in DeepEdit - See attached image
epistemic_AL

### CODE REVIEW

File: sample-apps/deepedit/main.py [hunk 1]
```diff
@@ -145,6 +152,13 @@ def init_strategies(self) -> Dict[str, Strategy]:
 
     def init_scoring_methods(self) -> Dict[str, ScoringMethod]:
         methods: Dict[str, ScoringMethod] = {}
+        if self.epistemic_enabled:
```
Comment (SachidanandAlle) [inline sample-apps/deepedit/main.py:155]: you need to add in init_strategies also

Comment (diazandr3s) [reply sample-apps/deepedit/main.py:158]: Added!

File: sample-apps/deepedit/main.py [hunk 2]
```diff
@@ -37,37 +39,39 @@
...
             ],
-            norm_name="instance",
-            deep_supervision=False,
-            res_block=True,
-        )
+            "norm_name": "instance",
+            "deep_supervision": False,
+            "res_block": True,
+        }
+        self.network = DynUNetV1(**network_params)
```
Comment (aihsani) [inline sample-apps/deepedit/main.py:73]: Is it really necessary to have tow network definitions here? Is it possible to use the same neural net but in `eval` mode in  other places:
e.g you could use
```
methods["EPISTEMIC"] = EpistemicScoring(
                model=[self.pretrained_model, self.final_model],
                network=self.network_with_dropout,
                transforms=self._infers["deepedit_seg"].pre_transforms(),
                num_samples=self.epistemic_samples,
            )
```
in scoring, but
```
def init_infers(self) -> Dict[str, InferTask]:
        return {
            "deepedit": Deepgrow(
                [self.pretrained_model, self.final_model],
                self.network,
                self.network.eval(),
                spatial_size=self.planner.spatial_size,
                target_spacing=self.planner.target_spacing,
            ),
            "Histogram+GraphCut": HistogramBasedGraphCut(),
        }
```

File: sample-apps/deepedit/main.py [hunk 3]
```diff
@@ -37,37 +39,39 @@
...
+        self.network_with_dropout = DynUNetV1(**network_params, dropout=0.2)
```
Comment (aihsani) [inline sample-apps/deepedit/main.py:74]: Are both of these trained independently? how do you know they are learning similar things? shouldn't the weights be shared so that they converge with and without dropout?

**Review by aihsani — APPROVED** [review]
